salt/config/__init__.py

```diff
@@ -2852,7 +2852,7 @@ def is_profile_configured(opts, provider, profile_name, vm_=None):
         non_size_drivers.append('linode')
 
     # If cloning on VMware, specifying image is not necessary.
-    if driver == 'vmware' and profile_key.get('image', True):
+    if driver == 'vmware' and 'image' not in list(profile_key.keys()) and 'clonefrom' in list(profile_key.keys()):
```

Comment:
It should be ``and clonefrom not in list(profile_key.keys()):`` instead of ``and 'clonefrom' in list(profile_key.keys()):``

1. If neither ``clonefrom`` nor ``image`` is specified, it should report an error that ``clonefrom`` is missing (that's the default)
2. If both ``image`` and ``clonefrom`` are specified, it should use ``clonefrom`` by default and ignore ``image``.